src/go/rpk/pkg/vcloud/login.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2021 Vectorized, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package vcloud
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	// audience of yak API in Auth0
+	// TODO(av) make it configurable to be able to work with production as well
+	yakAudience = "dev.vectorized.cloud"
+)
+
+var (
+	ErrTimeout = errors.New("authentication timed out")
+)
+
+type AuthToken struct {
+	Token string
+}
+
+// Login implements device flow of oauth2
+// reference https://auth0.com/docs/flows/call-your-api-using-the-device-authorization-flow
+func Login(auth0Client AuthClient) (*AuthToken, error) {
+	deviceCodeResponse, err := auth0Client.GetDeviceCode(yakAudience)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Info(fmt.Sprintf("Go to the following link in your browser (code: %s):\n", deviceCodeResponse.UserCode))
```

Comment:
I've seen some tools open the browser automatically, which saves the user one step. What do you think about that approach?

Reply:
yeah I think we could add that. Gcloud actually does both, while the one I implemented is NOT default. I actually like not opening browser better as a default but up for discussion. WDYT @senior7515

Reply:
we need auth to work on servers too, i'd say this would be fine as a first step, i think the browser might be a v2